lib/thor.rb

```diff
@@ -238,6 +238,9 @@ def subcommand(subcommand, subcommand_class)
         args.unshift("help") if opts.include? "--help" or opts.include? "-h"
         invoke subcommand_class, args, opts, :invoked_via_subcommand => true, :class_options => options
       end
+      subcommand_class.commands.each do |meth, command|
```

Comment:
@doudou thank you for your comment.

Did you mean like this?
```ruby
#!/usr/bin/env ruby

require 'thor'
require 'thor/version'
puts "Thor::VERSION: #{Thor::VERSION}"

module SubcommandTest1
  class Child1 < Thor
    namespace "child1"
    
    desc "foo NAME", "Fooo"
    def foo(name)
      puts "#{name} was given"
    end
  end
  
  class Parent < Thor
    desc "child1", "child1 description"
    subcommand "child1", Child1
  end
end

SubcommandTest1::Parent.start(ARGV)
```

Save this file as `thortest` and `chmod 755 thortest` .

But I think it behaves a little strange:
```
$ ./thortest             
Thor::VERSION: 0.19.1
Commands:
  thortest child1          # child1 description
  thortest help [COMMAND]  # Describe available commands or one specific command

$ ./thortest child1      
Thor::VERSION: 0.19.1
Commands:
  thortest child1 foo NAME        # Fooo
  thortest child1 help [COMMAND]  # Describe subcommands or one specific subc...

$ ./thortest child1 foo
Thor::VERSION: 0.19.1
ERROR: "thortest foo" was called with no arguments
Usage: "thortest foo NAME"
```

The namespace "child1" has gone from the last message.

What should I do to see the message like this?
```
$ ./thortest child1 foo
ERROR: "thortest child1 foo" was called with no arguments
Usage: "thortest child1 foo NAME"
```